### ISSUE TITLE
pdf2svg segfaults

### ISSUE THREAD

**Issue by doublex** [mozilla/pdf.js#8419]
Steps to reproduce (attached PDF):
[coredump.pdf](https://github.com/mozilla/pdf.js/files/1008487/coredump.pdf)

````
node examples/node/pdf2svg.js /tmp/coredump.pdf
````

Console:
````
# Document Loaded
Number of Pages: 2

# Page 1
Size: 308.88x606.72

Warning: Unimplemented graphic state BM
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
# Page 2
Size: 308.88x606.72

Warning: Unimplemented graphic state BM
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency

<--- Last few GCs --->

[30883:0x33e2030]    67959 ms: Scavenge 1373.3 (1414.7) -> 1373.3 (1414.7) MB, 0.3 / 0.0 ms  allocation failure incremental marking delaying mark-sweep
[30883:0x33e2030]    67977 ms: Mark-sweep 1373.3 (1414.7) -> 1147.4 (1176.9) MB, 18.8 / 0.0 ms  (+ 15.1 ms in 19 steps since start of marking, biggest step 5.8 ms, walltime since start of marking 4861 ms) last resort 
[30883:0x33e2030]    68032 ms: Mark-sweep 1147.4 (1176.9) -> 1147.1 (1152.4) MB, 54.9 / 0.0 ms  last resort 


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x1c67e52266a1 <JS Object>
    1: DoJoin(aka DoJoin) [native array.js:~97] [pc=0x2d223be6039b](this=0xbcbfd402311 <undefined>,q=0xfd62042e9f9 <JS Array[36]>,r=36,F=0xbcbfd4023b1 <true>,B=0xbcbfd402471 <String[0]: >,A=0xbcbfd402421 <false>)
    2: Join(aka Join) [native array.js:~122] [pc=0x2d223be6dfc2](this=0xbcbfd402311 <undefined>,q=0xfd62042e9f9 <JS Array[36]>,r=36,B=0xbcbfd402471 <String[0]: >,A=0xbcbfd402421 <false...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [node]
 2: 0x12b640c [node]
 3: v8::Utils::ReportOOMFailure(char const*, bool) [node]
 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [node]
 5: v8::internal::Factory::NewRawTwoByteString(int, v8::internal::PretenureFlag) [node]
 6: v8::internal::Runtime_StringBuilderConcat(int, v8::internal::Object**, v8::internal::Isolate*) [node]
 7: 0x2d223bb043a7
Aborted (core dumped)
````

**Comment by doublex** [comment]
$50 bounty:
https://www.bountysource.com/issues/45294148-pdf2svg-segfaults

**Comment by rishabhdugar** [comment]
Source of js script?

**Comment by yurydelendik** [comment]
@rishabhdugar 

STR:
1. clone pdf.js repo and install dependecies (see README.md)
2. run 'gulp dist'
3. save coredump.pdf
4. execute `node examples/node/pdf2svg.js coredump.pdf`

**Comment by doublex** [comment]
Could you apply this patch and close this issue?
https://github.com/yshahin/pdf.js/commit/b40b30bfb71b9f9008121dcb34a1da95d0e90242

**Comment by Rob--W** [comment]
This bug was reported again as #8534, and I submitted a PR with a proper fix at #8540
